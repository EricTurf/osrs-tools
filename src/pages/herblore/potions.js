const potions = {
    attackPotion: {
        herb: 'guam leaf',
        secondary: 'eye of newt',
        exp: 25,
        isDefault: true,
    },
    antipoison: {
        herb: 'marrentill',
        secondary: 'unicorn horn dust',
        exp: 37.5,
        isDefault: true,
    },
    strengthPotion: {
        herb: 'tarromin',
        secondary: 'limpwurt root',
        exp: 50,
    },
    serum207: {
        herb: 'tarromin',
        secondary: 'ashes',
        exp: 50,
        isDefault: true,
    },
    compostPotion: {
        herb: 'harralander',
        secondary: 'volcanic ash',
        exp: 60,
    },
    restorePotion: {
        herb: 'harralander',
        secondary: "red spiders' eggs",
        exp: 62.5,
    },
    energyPotion: {
        herb: 'harralander',
        secondary: 'chocolate dust',
        exp: 67.5,
        isDefault: true,
    },
    defencePotion: {
        herb: 'ranarr weed',
        secondary: 'white berries',
        exp: 75,
    },
    agilityPotion: {
        herb: 'toadflax',
        secondary: 'toads legs',
        exp: 80,
    },
    combatPotion: {
        herb: 'harralander',
        secondary: 'goat horn dust',
        exp: 84,
    },
    prayerPotion: {
        herb: 'ranarr weed',
        secondary: 'snape grass',
        exp: 87.5,
        isDefault: true,
    },
    superAttack: {
        herb: 'irit leaf',
        secondary: 'eye of newt',
        exp: 100,
        isDefault: true,
    },
    superantipoison: {
        herb: 'irit leaf',
        secondary: 'unicorn horn dust',
        exp: 106.3,
    },
    fishingPotion: {
        herb: 'avantoe',
        secondary: 'eye of newt',
        exp: 112.5,
    },
    superEnergy: {
        herb: 'avantoe',
        secondary: 'mort myre fungus',
        exp: 117.5,
        isDefault: true,
    },
    hunterPotion: {
        herb: 'avantoe',
        secondary: 'kebbit teeth dust',
        exp: 120,
    },
    superStrength: {
        herb: 'kwuarm',
        secondary: 'limpwurt root',
        exp: 125,
        isDefault: true,
    },
    weaponPoison: {
        herb: 'kwuarm',
        secondary: 'dragon scale dust',
        exp: 137.5,
    },
    superRestore: {
        herb: 'snapdragon',
        secondary: "red spiders' eggs",
        exp: 142.5,
        isDefault: true,
    },
    sanfewSerum: {
        herb: 'snapdragon',
        secondary: ["red spiders' eggs", 'unicorn horn dust'],
        isSpecial: true,
        exp: 160,
    },
    superDefence: {
        herb: 'cadantine',
        secondary: 'white berries',
        exp: 150,
        isDefault: true,
    },
    'antidote+': {
        herb: 'toadflax',
        secondary: 'yew root',
        base: 'coconut milk',
        exp: 155,
        isSpecial: true,
    },
    antifirePotion: {
        herb: 'lantadyme',
        secondary: 'dragon scale dust',
        exp: 157.5,
    },
    rangingPotion: {
        herb: 'dwarf weed',
        secondary: 'wine of zamorak',
        exp: 162.5,
        isDefault: true,
    },
    'weaponPoison+': {
        secondary: ['cactus spine', "red spiders' eggs"],
        herb: 'coconut milk',
        isSpecial: true,
        exp: 165,
    },
    magicPotion: {
        herb: 'lantadyme',
        secondary: 'potato cactus',
        exp: 172.5,
        isDefault: true,
    },
    staminaPotion: {
        herb: 'super energy potion',
        secondary: 'amylase crystal',
        isSpecial: true,
        exp: 25.5,
    },
    zamorakBrew: {
        herb: 'torstol',
        secondary: 'jangerberries',
        exp: 175,
    },
    'antidote++': {
        herb: 'irit leaf',
        base: 'coconut milk',
        secondary: 'magic roots',
        isSpecial: true,
        exp: 177.5,
    },
    bastionPotion: {
        herb: 'cadantine',
        base: 'vial of blood',
        secondary: 'wine of zamorak',
        isSpecial: true,
        exp: 155,
    },
    battlemagePotion: {
        herb: 'cadantine',
        base: 'vial of blood',
        secondary: 'potato cactus',
        isSpecial: true,
        exp: 155,
    },
    saradominBrew: {
        herb: 'toadflax',
        secondary: 'crushed nest',
        exp: 180,
        isDefault: true,
    },
    'weaponPoison++': {
        herb: 'nightshade',
        base: 'coconut milk',
        secondary: 'poison ivy berries',
        isSpecial: true,
        exp: 190,
    },
    extendedAntifire: {
        herb: 'antifire potion',
        secondary: 'lava scale shard',
        isSpecial: true,
        exp: 110,
    },
    antivenom: {
        herb: 'antidote++',
        secondary: 'zulrah scale (5)',
        isSpecial: true,
        exp: 120,
    },
    superCombatPotion: {
        herb: 'torstol',
        secondary: ['super attack(4)', 'super strength(4)', 'super defence(4)'],
        exp: 150,
        isDefault: true,
    },
    superAntifirePotion: {
        herb: 'antifire potion',
        secondary: 'crushed superior dragon bones',
        isSpecial: true,
        exp: 130,
    },
    'antivenom+': {
        herb: 'torstol',
        secondary: 'anti venom',
        isSpecial: true,
        exp: 125,
    },
    extendedSuperAntifire: {
        herb: 'super antifire',
        secondary: 'lava scale shard',
        isSpecial: true,
        exp: 160,
    },
};

export default potions;
