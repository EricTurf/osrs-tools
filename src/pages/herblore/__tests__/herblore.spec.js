import React from 'react';
import { render, fireEvent } from 'react-testing-library';

import potions from '../potions';

import HerbloreCalculator from '../herblore';

describe('Herblore Calculator', () => {
    describe('rendering', () => {
        it('should match the default snapshot', () => {
            const { container } = render(<HerbloreCalculator />);

            expect(container).toMatchSnapshot();
        });
    });
    describe('calculation', () => {
        it('should properly calculate herblore exp', () => {
            const { attackPotion } = potions;

            const guamAmount = 25;
            const expectedExpGained = 25 * attackPotion.exp;

            const { container, getByText, getByTestId } = render(<HerbloreCalculator />);
            const guamInput = container.querySelector(`input[name="${attackPotion.herb}"]`);

            fireEvent.change(guamInput, { target: { value: guamAmount } });

            const expGainedEl = getByTestId('totalExp');
            expect(expGainedEl.innerHTML).toEqual(`In total you will gain ${expectedExpGained} herblore exp.`);
        });
    });
});
