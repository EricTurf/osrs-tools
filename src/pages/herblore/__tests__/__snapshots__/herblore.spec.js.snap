// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Herblore Calculator rendering should match the default snapshot 1`] = `
<div>
  <div
    class="sc-htoDjs bZzhDu"
  >
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/f/fd/Guam_leaf.png?8fae5"
        title="guam leaf"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="guam leaf"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/2/25/Eye_of_newt.png?cf257"
          title="eye of newt"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/8/86/Attack_potion%283%29.png?74c44"
          title="attackPotion"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/b/ba/Marrentill.png?4fb2b"
        title="marrentill"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="marrentill"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/1/14/Unicorn_horn_dust.png?26485"
          title="unicorn horn dust"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/7/7b/Antipoison%283%29.png?02e60"
          title="antipoison"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/7/78/Tarromin.png?ada2a"
        title="tarromin"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="tarromin"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/f/f9/Ashes.png?a6f45"
          title="ashes"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/8/8c/Serum_207_%283%29.png?d05f0"
          title="serum207"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/c/cb/Harralander.png?bdd9c"
        title="harralander"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="harralander"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/0/0c/Chocolate_dust.png?b1f54"
          title="chocolate dust"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/6/6a/Energy_potion%283%29.png?29afb"
          title="energyPotion"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/a/a5/Ranarr_weed.png?ada2a"
        title="ranarr weed"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="ranarr weed"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/3/30/Snape_grass.png?c7b1f"
          title="snape grass"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/0/07/Prayer_potion%283%29.png?f9b25"
          title="prayerPotion"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/a/ae/Irit_leaf.png?3ed63"
        title="irit leaf"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="irit leaf"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/2/25/Eye_of_newt.png?cf257"
          title="eye of newt"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/b/bc/Super_attack%283%29.png?e48de"
          title="superAttack"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/9/9e/Avantoe.png?8fae5"
        title="avantoe"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="avantoe"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/2/24/Mort_myre_fungus.png?e4571"
          title="mort myre fungus"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/a/ab/Super_energy%283%29.png?84f18"
          title="superEnergy"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/c/c6/Kwuarm.png?3ed63"
        title="kwuarm"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="kwuarm"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/1/18/Limpwurt_root.png?fe88f"
          title="limpwurt root"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/9/9f/Super_strength%283%29.png?25714"
          title="superStrength"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/4/4c/Cadantine.png?8fae5"
        title="cadantine"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="cadantine"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/5/56/White_berries.png?e2e46"
          title="white berries"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/9/99/Super_defence%283%29.png?3c0ef"
          title="superDefence"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/2/23/Snapdragon.png?ada2a"
        title="snapdragon"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="snapdragon"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/8/8e/Red_spiders%27_eggs.png?dab74"
          title="red spiders' eggs"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/d/dc/Super_restore%283%29.png?b14cf"
          title="superRestore"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/a/ad/Toadflax.png?14698"
        title="toadflax"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="toadflax"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/e/e6/Crushed_nest.png?7a5b0"
          title="crushed nest"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/b/b6/Saradomin_brew%283%29.png?8ae4e"
          title="saradominBrew"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/8/82/Lantadyme.png?3ed63"
        title="lantadyme"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="lantadyme"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/5/56/Potato_cactus.png?fb31b"
          title="potato cactus"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/0/02/Magic_potion%283%29.png?774cb"
          title="magicPotion"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/2/22/Dwarf_weed.png?8fae5"
        title="dwarf weed"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="dwarf weed"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/c/ca/Wine_of_zamorak.png?4819a"
          title="wine of zamorak"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/0/0f/Ranging_potion%283%29.png?b09a6"
          title="rangingPotion"
        />
      </div>
    </div>
    <div
      class="sc-EHOje epUP"
    >
      <img
        class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
        src="https://oldschool.runescape.wiki/images/a/a4/Torstol.png?14698"
        title="torstol"
      />
      <div
        class="sc-bdVaJa kVMMIv"
        value=""
      >
        <input
          class="sc-htpNat dlPGwY"
          name="torstol"
          type="text"
          value=""
        />
        <span
          class="sc-bwzfXH kjMOwR"
        />
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/f/ff/Super_attack%284%29.png?88fac"
          title="super attack(4)"
        />
         
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/3/32/Super_strength%284%29.png?a035c"
          title="super strength(4)"
        />
         
        +
        0
         
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/9/94/Super_defence%284%29.png?22916"
          title="super defence(4)"
        />
         
      </div>
      <div
        class="sc-bZQynM dnFZLg"
      >
        = 
        0
         exp and 
        0
        <img
          class="sc-ifAKCX zfpgZ sc-bxivhb iwGdkg"
          src="https://oldschool.runescape.wiki/images/9/94/Super_combat_potion%283%29.png?27de0"
          title="superCombatPotion"
        />
      </div>
    </div>
    <p
      class="sc-gzVnrw kUlmDW"
      data-testid="totalExp"
    >
      In total you will gain 
      0
       herblore exp.
    </p>
  </div>
</div>
`;
